GVFS/GVFS.UnitTests/Virtualization/Projection/SortedFolderEntriesTests.cs

```diff
@@ -117,15 +117,45 @@ public void ValidateOrderOfDefaultEntries()
         }
 
         [TestCase]
-        public void AddItemWithIsIncludedFalse()
+        public void AddFolderNoIncludeChildrenShouldBeIncluded()
         {
             SortedFolderEntries sfe = SetupDefaultEntries();
             LazyUTF8String name = ConstructLazyUTF8String("IsIncludedFalse");
-            sfe.AddFile(name, new byte[20], isIncluded: false);
-            sfe.TryGetValue(name, out FolderEntryData folderEntryData).ShouldBeTrue();
-            folderEntryData.ShouldNotBeNull();
-            folderEntryData.IsFolder.ShouldBeFalse();
-            folderEntryData.IsIncluded.ShouldBeFalse();
+            sfe.GetOrAddFolder(new[] { name }, partIndex: 0, parentIsIncluded: false, rootSparseFolderData: new SparseFolderData());
+            ValidateFolder(sfe, name, isIncludedValue: true);
+        }
+
+        [TestCase]
+        public void AddFolderWhereParentIncludedIsFalseAndIncluded()
+        {
+            SortedFolderEntries sfe = SetupDefaultEntries();
+            LazyUTF8String name = ConstructLazyUTF8String("Child");
+            SparseFolderData sparseFolderData = new SparseFolderData();
+            sparseFolderData.Children.Add("Child", new SparseFolderData());
+            sfe.GetOrAddFolder(new[] { name }, partIndex: 0, parentIsIncluded: false, rootSparseFolderData: sparseFolderData);
+            ValidateFolder(sfe, name, isIncludedValue: false);
+        }
+
+        [TestCase]
+        public void AddFolderWhereParentIncludedIsTrueAndChildIsIncluded()
+        {
+            SortedFolderEntries sfe = SetupDefaultEntries();
+            LazyUTF8String name = ConstructLazyUTF8String("Child");
+            SparseFolderData sparseFolderData = new SparseFolderData();
+            sparseFolderData.Children.Add("Child", new SparseFolderData());
+            sfe.GetOrAddFolder(new[] { name }, partIndex: 0, parentIsIncluded: true, rootSparseFolderData: sparseFolderData);
+            ValidateFolder(sfe, name, isIncludedValue: true);
+        }
+
+        [TestCase]
+        public void AddFolderWhereParentIncludedIsTrueAndChildIsNotIncluded()
+        {
+            SortedFolderEntries sfe = SetupDefaultEntries();
+            LazyUTF8String name = ConstructLazyUTF8String("ChildNotIncluded");
+            SparseFolderData sparseFolderData = new SparseFolderData();
+            sparseFolderData.Children.Add("Child", new SparseFolderData());
+            sfe.GetOrAddFolder(new[] { name }, partIndex: 0, parentIsIncluded: true, rootSparseFolderData: sparseFolderData);
+            ValidateFolder(sfe, name, isIncludedValue: false);

```

Comment:
Is there a test to handle the case where a parent folder is recursively included?